functions/nfts/asset/[[index]].ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable import/no-unused-modules */
+import { MetaTagInjector } from '../../components/metaTagInjector'
+import getAsset from '../../utils/getAsset'
+
+export const onRequest: PagesFunction = async ({ params, request, next }) => {
+  const { index } = params
+  const collectionAddress = String(index[0])
+  const tokenId = String(index[1])
+  const assetPromise = getAsset(collectionAddress, tokenId, request.url)
+  const resPromise = next()
+  try {
+    const [data, res] = await Promise.all([assetPromise, resPromise])
```

Comment:
Call it `asset` to match the `assetPromise`.
```suggestion
    const [asset, res] = await Promise.all([assetPromise, resPromise])
```